subsys/bluetooth/audio/ascs.c

```diff
@@ -1378,6 +1377,58 @@ static int ase_config(struct bt_ascs *ascs, struct bt_ascs_ase *ase,
 	return 0;
 }
 
+int ascs_ase_config(struct bt_conn *conn, struct bt_audio_ep *ep,
+					struct bt_codec *codec)
+{
+	struct bt_codec old_codec;
+	struct bt_audio_stream *stream;
+	int err;
+
+	/* Store current codec configuration to be able to restore it
+	 * in case of error.
+	 */
+	(void)memcpy(&old_codec, &ep->codec, sizeof(old_codec));
+
+	if (ascs_ep_set_codec(ep, codec->id,
+				sys_le16_to_cpu(codec->cid),
+				sys_le16_to_cpu(codec->vid),
+				NULL, 0, codec)) {
+		(void)memcpy(&ep->codec, &old_codec, sizeof(old_codec));
+		return -ENOTSUP;
+	}
+
+	stream = NULL;
+	if (unicast_server_cb != NULL &&
+		unicast_server_cb->config != NULL) {
+		err = unicast_server_cb->config(conn, ep,
+						ep->dir,
+						&ep->codec, &stream,
+						&ep->qos_pref);
+	} else {
+		err = -ENOTSUP;
+	}
```

Comment:
If the server is configuring a stream via `bt_audio_stream_config` I think we should skip the callback here, and have the server supply the `stream` in the function call instead.